File: BOJ/2606.c
// 2606 바이러스
#include <stdio.h>

int cnt = 0;
int adj[101][101] = {0, };
int visited[101] = {0, };

void DFS(int n, int m, int cur) {
    for (int i = 1; i <= n; ++i) {
        if (visited[i] == 0 && adj[cur][i] == 1) {
            visited[i] = 1;
            cnt++;
            DFS(n, m, i);
        }
    }

    return;
}

int main(void)
{
    int n, m, a, b;

    scanf("%d %d", &n, &m);

    for (int i = 0; i < m; ++i) {
        scanf("%d %d", &a, &b);
        adj[a][b] = adj[b][a] = 1;    
    }
    visited[1] = 1;
    DFS(n, m, 1);
    printf("%d", cnt);
    return 0;
}

File: BOJ/2606.py
# 2606 바이러스
import sys
input = sys.stdin.readline
sys.setrecursionlimit(10 ** 7)

node = int(input())
edge = int(input())

adj = [ [] for _ in range(node + 1)]
visited = [False] * (node + 1)
answer = 0

for _ in range(edge):
    a, b = map(int, input().split())
    adj[a].append(b)
    adj[b].append(a)
    
def DFS(cur):
    global answer
    visited[cur] = True
    
    for next in adj[cur]:
        if visited[next] == False:
            answer += 1 # 1번 자기 자신은 제외
            DFS(next)
            
DFS(1)
print(answer)